mne/channels/channels.py

```diff
@@ -1070,6 +1070,11 @@ def add_channels(self, add_list, force_update_info=False):
             self._read_picks = [
                 np.concatenate([r, extra_idx]) for r in self._read_picks]
             assert all(len(r) == self.info['nchan'] for r in self._read_picks)
+        elif isinstance(self, BaseEpochs):
+            self.picks = np.arange(self._data.shape[1])
+            if hasattr(self, '_projector'):
+                del self._projector
```

Comment:
I suspect that this will be a problem with delayed projection. Probably better to do some variant of what's done in `__init__`:
```
        activate = False if self._do_delayed_proj else proj
        self._projector, self.info = setup_proj(self.info, False,
                                                activate=activate)
```
and also add a regression test that would have caught this error (e.g., use delayed projection, get data, add channel, get data again, ensure they match -- which they shouldn't until you make the above suggested change).